File: leetcode/exercises/1.3.1.easy.valid_anagram_3.go
// -----------------------
// Created     : 30/10/2025
// Last Edited : 30/10/2025
// Topics      : 
// Big O       :
// Problem Id  : 242
// References  :
// Notes       :
// -----------------------

package main

import (
    "fmt"
    "sort"
)

func isAnagram(s, t string) bool {
    sr := []rune(s)
    tr := []rune(t)

    sort.Slice(sr, func (i, j int) bool {
        return sr[i] < sr[j]
    })    
    sort.Slice(tr, func (i, j int) bool {
        return tr[i] < tr[j]
    })    

    return string(sr) == string(tr)
}

func main() {
    fmt.Println(isAnagram("oiii", "io"))
}

File: leetcode/exercises/1.3.1.easy.valid_anagram_3.js
// -----------------------
// Created     : 24/10/2025
// Last Edited : 24/10/2025
// Topics      : 
// Big O       : 
// Problem Id  : 242
// References  :
// Notes       :
// -----------------------

/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
    return s.split("").sort().join("") === t.split("").sort().join("");
};

console.log(isAnagram("economia", "enocomia"));


